### ISSUE TITLE
Duplicate Device Added After Upgrade and old entities stop working

### ISSUE THREAD

**Issue by jstride** [travisghansen/hass-pfsense#99]
![image](https://user-images.githubusercontent.com/1501958/184516207-9d29f8d4-f35a-4f7e-bb06-5ed026aca96d.png)

After upgrading a dupliate entry is added (the location is blank compared to the original) and all entitiies are duplicated with _2 at the end. The original entities are no longer reported.

**Comment by travisghansen** [comment]
From what version to what version? Did you happen to reinstall pfsense?

**Comment by jstride** [comment]
I'm on v0.5.7 - upgrade was from the last minor version as I upgrade regularly. Only thing that has changed AFAIK is I rebooted the router and connected it to new switch via DAC rather than fibre module so one of the primary interfaces will have had a different speed.

I did delete, restart HASS, then re-add the router a week ago but that didn't create a duplicate at the time, the duplicate has appeared since the reboot of the router.

**Comment by travisghansen** [comment]
I’m wondering if the netgate id of the device changed. That unique id is used by the integration and if it changes then that would explain new entities.

Can you just remove the integration and re-add to make them all reappear without any duplicates?

**Comment by jstride** [comment]
Yes, removing and re-adding does remove the duplicate. I had done this previously when a duplicate had occurred.

This has resolved the issue, but still don't understand the cause. Will keep an eye our for if it happens again. As an FYI 0.57->0.58 was offered so I did the plugin upgrade after removing the integrations and before re-adding.
________________________________

**Comment by travisghansen** [comment]
Yeah, make note of your netgate device id and let's observe if it changes in the future.

**Comment by nagyrobi** [comment]
Still happening for me. After 3 consecutive upgrades, and now after replacing the pfSense hardware (same config restored), I have 4 instances in Home Assistant...

**Comment by thrawnarn** [comment]
I have the same problem and I can confirm that the netgate device id has changed between my 3 registred devices (in HA). Could it be that I run a virtualized version of pfsense?

**Comment by wymangr** [comment]
I have this problem also. For me, I enabled a sensor, refreshed and the original device showed all the entities "Unavailable" and it created a second device. I've disabled the not working device, is there a way to at least remove it?
